rcl/test/rcl/test_subscription.cpp

```diff
@@ -77,7 +77,34 @@ class CLASSNAME (TestSubscriptionFixture, RMW_IMPLEMENTATION) : public ::testing
   }
 };
 
-void
+rcl_ret_t
+wait_for_established_subscription(
+  const rcl_publisher_t * publisher,
+  size_t max_tries,
+  int64_t period_ms,
+  bool & success)
+{
+  size_t iteration = 0;
+  rcl_ret_t ret = RCL_RET_OK;
+  size_t subscription_count = 0;
+  while (iteration < max_tries) {
+    ++iteration;
+    ret = rcl_publisher_get_subscription_count(publisher, &subscription_count);
+    if (ret != RCL_RET_OK) {
+      success = false;
+      return ret;
+    }
+    if (subscription_count == 1) {
+      success = true;
+      return ret;
+    }
+    std::this_thread::sleep_for(std::chrono::milliseconds(period_ms));
+  }
+  success = false;
+  return ret;
```

Comment:
@Barry-Xu-2018 I still can't fathom why you need an additional boolean, but others may think otherwise.

Reply:
I agree with @hidmic; we can just use the `rcl_ret_t` of the function.  No need for the additional `success` variable, it doesn't really add any new information.  This also goes for `wait_for_server_to_be_available` above.

Reply:
Thanks.  Update codes.